commands/system/help.js

```diff
@@ -16,67 +17,98 @@ class Help extends Command {
-      const output = sorted.filter(c => !(level < 10 && c.help.category == "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map(c => {
-        const cat = c.help.category.toProperCase();
-        if (currentCategory !== cat && !type) {
-          currentCategory = cat;
-          return `\n\`${message.settings.prefix}help ${cat.toLowerCase()}\` | Shows ${cat} commands`;
-        }
-      }).join("");
+    let embed;
+    const embedPreset = {
+      timestamp: new Date().getTime(),
+      color: message.guild.me.roles.highest.color || 5198940,
+      footer: {
+        text: `Requested by ${message.author.tag}`,
+        iconURL: message.author.avatarURL()
+      }
+    };
 
-      embed.setDescription(description)
-        .addField("Categories", output);
+    if (this.client.commands.has(type) || this.client.commands.some(command => command.conf.aliases.includes(type))) {
+      const cm = this.client.commands.get(type) || this.client.commands.get(this.client.aliases.get(type));
+      if (level < this.client.levelCache[cm.conf.permLevel]) return;
+      embed = new MessageEmbed(embedPreset)
+        .setTitle(cm.help.name.toProperCase())
+        .setDescription(`\u200B\t${cm.help.description}`)
+        .addField("Command details", cm.help.extended, false)
+        .addField("Command usage", `\`${cm.help.usage}\``, false)
+        .addField("Command aliases", cm.conf.aliases.length == 0 ? "None" : cm.conf.aliases.join(", "), false);
+      
+      return message.channel.send({ embed });
     } else {
-      let n = 0;
-      sorted.forEach(c => {
-        if (c.help.category.toLowerCase() === type.toLowerCase()) {
-          n++;
+      let currentCategory = "";
+      const sorted = this.client.commands.sort((p, c) => p.help.category > c.help.category ? 1 :  p.help.name > c.help.name && p.help.category === c.help.category ? 1 : -1 );
+      embed = new PaginationEmbed(embedPreset)
+        .setAuthorisedUser(message.author)
+        .setChannel(message.channel)
+        .setElementsPerPage(perpage)
+        .showPageIndicator(true)
+        .setTimeout(60 * 1000);
+      
+      const isPage = !isNaN(parseInt(type));
+      const isDefault = !type || isPage;
+
+      if (isDefault) {
+        const output = [];
+        const title = "Command category list";
+        const description = `Use \`${message.settings.prefix}help <category>\` to find commands for a specific category`;
+        const tip = `\nTo skip to a page for more categories:\n\tReact with ↗; or\n\tDo \`${message.settings.prefix}help <page-num>\``;
+        const categories = sorted.filter(c => !(level < 10 && c.help.category == "Owner") || !(c.help.category === "NSFW" && !message.channel.nsfw)).map(c => {
+          const cat = c.help.category.toProperCase();
+          if (currentCategory !== cat && isDefault) {
```

Comment:
You could avoid this and the latter processing by filtering it inside the `sorted.filter()`.